### ISSUE TITLE
Error occured when add ultisnip source into nvim-cmp

### ISSUE THREAD

**Issue by hiberabyss** [quangnguyen30192/cmp-nvim-ultisnips#23]
Following error will occur when edit cpp file with snip completion:

![image](https://user-images.githubusercontent.com/2898670/143447959-260e771e-f20b-4b3a-945f-36016ccd12ed.png)

neovim version: NVIM v0.6.0-dev+627-gb5b025f6a

**Comment by smjonas** [comment]
Hey, sorry about that. Can you share your snippet definitions for .cpp files?

**Comment by PetitBandit** [comment]
<img width="1792" alt="CleanShot 2021-11-25 at 17 14 53@2x" src="https://user-images.githubusercontent.com/32488040/143474737-68b6dccd-610f-4c4e-9af6-a05ce2ed25bc.png">


hey having this issue since i Upgraded this morning .
I m unsure that this is the same issue.

**Comment by PetitBandit** [comment]
![CleanShot 2021-11-25 at 17 17 26@2x](https://user-images.githubusercontent.com/32488040/143474993-2dea2969-cba8-4bee-b208-ac3aa6b36ef1.png)


I have a similar issue since i upgraded plugins this morning. weirdly, some snippet still work, but most of them arent.
I dont have enough knowledge to figure out if it s the same issue or not.

**Comment by hiberabyss** [comment]
> 

Seems there will be problem when `honza/vim-snippets.git` installed

**Comment by hiberabyss** [comment]
Have another test with any custom snippets for cpp, there will be the error even `honza/vim-snippets.git` not installed.

**Comment by smjonas** [comment]
@hiberabyss thanks for the info, I will try to reproduce it and push a fix later

@PetitBandit it probably is the same issue, I recently wrote a parser for snippet headers (with insufficient tests as it seems)

**Comment by smjonas** [comment]
@PetitBandit Unfortunately, I am not able to reproduce the issue using the typescriptreact snippets from your dotfiles repo (please ignore the duplicates).

![cmpu_issue23](https://user-images.githubusercontent.com/40792180/143497105-5d7a6155-d1de-443e-854f-b44542be890a.png)

I assume this was the file you were using for your snippets? Or are you using any
additional snippet files? To find out why this is happening, I really need to know the exact
snippet that causes problems during parsing.

**Comment by smjonas** [comment]
@hiberabyss I could not reproduce this with a custom snippet definition in a `cpp.snippets` file. Could you share an example snippet that caused the issues for you?

**Comment by smjonas** [comment]
I was finally able to reproduce using the `c.snippets` file from honza/vim-snippets :)

**Comment by quangnguyen30192** [comment]
@hiberabyss the issue was fixed could you please update the plugin and try again?

**Comment by hiberabyss** [comment]
This error was fixed. Thanks!

However, seems some snippets like  `once` in `honza/vim-snippets` could not be completed.

**Comment by quangnguyen30192** [comment]
Thanks, we'll test thoroughly for the others and improve the implementation.

**Comment by PetitBandit** [comment]
Hey, I updated again the plugin, but the error continues. 
i can reproduce the issue with the snippet:

```
snippet bgct "tomato for the win"
backgroundColor: 'tomato', //TODO: petit bandit <-- remove this
endsnippet
```

**Comment by smjonas** [comment]
@PetitBandit should be fixed in the latest PR (#26). Sorry for all the breaking changes that should not have been breaking ^^

**Comment by PetitBandit** [comment]
The error was fixed, thank you so much for this quickfix, much appreciated !
